In [2]:
import pandas as pd 
a=[2,34,5,5,55,3]
myvar=pd.Series(a)
print(myvar)
#with series we defien a column just one column

0     2
1    34
2     5
3     5
4    55
5     3
dtype: int64


In [3]:
import pandas as pd

calories = {"day1": 420, "day2": 380, "day3": 390}

myvar = pd.Series(calories)

print(myvar)

day1    420
day2    380
day3    390
dtype: int64


In [13]:
# series is like a coluam a datafreame si the whle table 

data ={"day1":[123,23423,23434,242,23,23,23,23,2,32],
       "day2":[2,243,4,3,234,234,2,423,423,423]
      }
myvar = pd.DataFrame(data)

print(myvar)
print(myvar.loc[0])

    day1  day2
0    123     2
1  23423   243
2  23434     4
3    242     3
4     23   234
5     23   234
6     23     2
7     23   423
8      2   423
9     32   423
day1    123
day2      2
Name: 0, dtype: int64


In [14]:
# Read CSV files in pandas 
import pandas as pd 
df=pd.read_csv('data.csv')

print(df.to_string())
#use to_string() to print the entire DataFrame.

     Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
5          60    102       127     300.0
6          60    110       136     374.0
7          45    104       134     253.3
8          30    109       133     195.1
9          60     98       124     269.0
10         60    103       147     329.3
11         60    100       120     250.7
12         60    106       128     345.3
13         60    104       132     379.3
14         60     98       123     275.0
15         60     98       120     215.2
16         60    100       120     300.0
17         45     90       112       NaN
18         60    103       123     323.0
19         45     97       125     243.0
20         60    108       131     364.2
21         45    100       119     282.0
22         60    130       101     300.0
23         45   

In [32]:
import pandas as pd 


df=pd.read_csv('data.csv')

print(df.head())
print(df.tail()) 
#The head() method returns the headers and a specified number of rows, starting from the top.
# print out the first few columss

   Duration  Pulse  Maxpulse  Calories
0        60    110       130     409.1
1        60    117       145     479.0
2        60    103       135     340.0
3        45    109       175     282.4
4        45    117       148     406.0
     Duration  Pulse  Maxpulse  Calories
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4


In [25]:
import pandas as pd

df = pd.read_json('data.json')

print(df.to_string()) 


     Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
5          60    102       127     300.5
6          60    110       136     374.0
7          45    104       134     253.3
8          30    109       133     195.1
9          60     98       124     269.0
10         60    103       147     329.3
11         60    100       120     250.7
12         60    106       128     345.3
13         60    104       132     379.3
14         60     98       123     275.0
15         60     98       120     215.2
16         60    100       120     300.0
17         45     90       112       NaN
18         60    103       123     323.0
19         45     97       125     243.0
20         60    108       131     364.2
21         45    100       119     282.0
22         60    130       101     300.0
23         45   

In [26]:
#JSON = Python Dictionary

#JSON objects have the same format as Python dictionaries.

In [33]:
#Data cleaning means fixing bad data in your data set.

#Bad data could be:

# 1.empty cells , 2.data in wrongformat 3.wrong data , 4: duplicates

In [35]:
# remove the rows 
#Return a new Data Frame with no empty cells:
import pandas as pd

df = pd.read_csv('data.csv')

new_df = df.dropna()

print(new_df.to_string())

     Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
5          60    102       127     300.0
6          60    110       136     374.0
7          45    104       134     253.3
8          30    109       133     195.1
9          60     98       124     269.0
10         60    103       147     329.3
11         60    100       120     250.7
12         60    106       128     345.3
13         60    104       132     379.3
14         60     98       123     275.0
15         60     98       120     215.2
16         60    100       120     300.0
18         60    103       123     323.0
19         45     97       125     243.0
20         60    108       131     364.2
21         45    100       119     282.0
22         60    130       101     300.0
23         45    105       132     246.0
24         60   

In [42]:
# cells with data of worng farmat .
#comvert into a correct format 
import pandas as pd

df = pd.read_csv('data.csv')

df['Calories'] = pd.to_datetime(df['Calories'], format='mixed')

print(df.to_string())

     Duration  Pulse  Maxpulse                      Calories
0          60    110       130 1970-01-01 00:00:00.000000409
1          60    117       145 1970-01-01 00:00:00.000000479
2          60    103       135 1970-01-01 00:00:00.000000340
3          45    109       175 1970-01-01 00:00:00.000000282
4          45    117       148 1970-01-01 00:00:00.000000406
5          60    102       127 1970-01-01 00:00:00.000000300
6          60    110       136 1970-01-01 00:00:00.000000374
7          45    104       134 1970-01-01 00:00:00.000000253
8          30    109       133 1970-01-01 00:00:00.000000195
9          60     98       124 1970-01-01 00:00:00.000000269
10         60    103       147 1970-01-01 00:00:00.000000329
11         60    100       120 1970-01-01 00:00:00.000000250
12         60    106       128 1970-01-01 00:00:00.000000345
13         60    104       132 1970-01-01 00:00:00.000000379
14         60     98       123 1970-01-01 00:00:00.000000275
15         60     98    

In [45]:
# delete the nah / null cells 
import pandas as pd

df = pd.read_json('data.json')
df.dropna(subset=['Calories'], inplace = True)

print(df.to_string())

     Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
5          60    102       127     300.5
6          60    110       136     374.0
7          45    104       134     253.3
8          30    109       133     195.1
9          60     98       124     269.0
10         60    103       147     329.3
11         60    100       120     250.7
12         60    106       128     345.3
13         60    104       132     379.3
14         60     98       123     275.0
15         60     98       120     215.2
16         60    100       120     300.0
18         60    103       123     323.0
19         45     97       125     243.0
20         60    108       131     364.2
21         45    100       119     282.0
22         60    130       101     300.0
23         45    105       132     246.0
24         60   

# cleaning the worng data
#"Wrong data" does not have to be "empty cells" or "wrong format", it can just be wrong, like if someone registered "199" instead of "1.99".
# the worng data can be remove by manually
df.loc[7,'Duration']=45
#Loop through all values in the "Duration" column.
for x in df.index:
  if df.loc[x, "Duration"] > 120:
    df.loc[x, "Duration"] = 120

In [47]:
#Duplicate rows are rows that have been registered more than one time.duplicated() method.
import pandas as pd

df = pd.read_csv('data.csv')
print(df.duplicated())

0      False
1      False
2      False
3      False
4      False
       ...  
164    False
165    False
166    False
167    False
168    False
Length: 169, dtype: bool


In [51]:
import pandas as pd

df = pd.read_csv('data.csv')
df.drop_duplicates(inplace = True)
print(df)
# remove all the duplicates

     Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
..        ...    ...       ...       ...
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4

[162 rows x 4 columns]


In [ ]:
#The corr() method calculates the relationship between each column in your data set.